src/lib/core/TLVUtilities.cpp

```diff
@@ -33,6 +33,14 @@ namespace TLV {
 
 namespace Utilities {
 
+namespace {
+
+// Sets up a limit on recursion depth, to avoid any stack overflows
+// on very deep TLV structures. Embedded has limited stack space.
+constexpr size_t kMaxRecursionDepth = 10;
```

Comment:
Wrote a script to show the deepest structures in Controller-clusters right now. I got max of 7:

```
...
6 EnhancedViewSceneResponse::extensionFieldSets[].attributeValueList[]
6 TestListNestedStructListArgumentRequestRequest::arg1[].d[]
6 TestStructArrayArgumentRequestRequest::arg1[].d[]
6 TestStructArrayArgumentResponse::arg1[].d[]
6 ViewSceneResponse::extensionFieldSets[].attributeValueList[]
7 AddSceneRequest::extensionFieldSets[].attributeValueList[].attributeValue[]
7 EnhancedAddSceneRequest::extensionFieldSets[].attributeValueList[].attributeValue[]
7 EnhancedViewSceneResponse::extensionFieldSets[].attributeValueList[].attributeValue[]
7 ViewSceneResponse::extensionFieldSets[].attributeValueList[].attributeValue[]
```

We may need a bit extra wrapping for responsedata wrapping.